components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -5227,6 +5234,55 @@ public boolean isConfigured() {
         return certificateManager.getCertificates(tenantId);
     }
 
+    @Override
+    public List<CertificateMetadataDTO> searchCertificates(int tenantId, String alias, String endpoint) throws
+            APIManagementException {
+
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        return certificateManager.getCertificates(tenantId, alias, endpoint);
+    }
+
+    @Override
+    public boolean isCertificatePresent(int tenantId, String alias) throws APIManagementException {
+
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        return certificateManager.isCertificatePresent(tenantId, alias);
+    }
+
+    @Override
+    public CertificateInformationDTO getCertificateStatus(String alias) throws APIManagementException {
+
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        return certificateManager.getCertificateInformation(alias);
+    }
+
+    @Override
+    public int updateCertificate(String certificateString, String alias) throws APIManagementException {
+
+        CertificateManager certificateManager = new CertificateManagerImpl();
+        ResponseCode responseCode = certificateManager.updateCertificate(certificateString, alias);
+
+        if (ResponseCode.SUCCESS == responseCode) {
```

Comment:
Should this be if(ResponseCode.SUCCESS == responseCode.getResponseCode())?

Reply:
ResponseCode.SUCCESS is ResponseCode type. responseCode.getResponseCode() is int type.

responseCode will only be null if an exception is thrown from the certificateManager.updateCertificate method. If so, the if condition will not be executed as the ApiProvider updateCertificate, throws the exception to the upper layer.